### ISSUE TITLE
Add ability to use thumbnails for icons for better visuals to sites

### ISSUE THREAD

**Issue by gtanime** [zombieFox/nightTab#170]
Add ability to use thumbnails for icons for better visuals to sites. my kid uses the comp a lot and hes 5 so  reading isnt his strong-suit yet

**Comment by zombieFox** [comment]
Hi, unfortunately images or thumbnails are out of scope of this project. Since the project does not store any information on a database there is a limit to the amount of data in the browsers cache for extensions or sites. Allowing website thumbnails or images would very likely exceed the local storage limits. Please also see #100.

nightTab does however support icons from the fontawesome project. Each bookmark can show an icon in place of a letter using the bookmark edit modal:

<img width="589" alt="Screen Shot 2020-04-04 at 23 51 01" src="https://user-images.githubusercontent.com/7115017/78462944-24724d80-76cf-11ea-9a3b-82e89938f40d.png">

**Comment by Excigma** [comment]
In my opinion the font awesome icons are enough.

But how about loading icons/images from a URL?

**Comment by zombieFox** [comment]
> But how about loading icons/images from a URL?

This could be an option. I'll investigate.

**Comment by mutageneral** [comment]
> But how about loading icons/images from a URL?

I would use that feature if it was there as I really prefer good looking icons that match one of the websites than text or an icon that kinda makes you think of the website. For example, with my bookmark to the arch wiki, I had to put a literal arch! 😂

Even though the icons wouldn't be automatically themed, I think I would figure out how to retheme one into the right colour and then upload it somewhere to use it.

**Comment by StormPlayzz** [comment]
> But how about loading icons/images from a URL?

Would also love this! I have 50+ Bookmarks and only like 5 of them have an icon from fontawesome. It's becoming very hard to find certain sites now... Loading icons from another source would be amazing!

**Comment by zombieFox** [comment]
> Loading icons from another source would be amazing!

This feature would very likely be loading an image file from an outside source. So the hosting, appearance, size and transparacy of the icon/image would rely on the user to manage. 

Still investigating. 🌵 💻

**Comment by zombieFox** [comment]
Turns out adding support for bookmark images also adds support for animated gifs 👾 

![Kapture 2020-06-07 at 13 23 04](https://user-images.githubusercontent.com/7115017/83968593-2c628f80-a8c2-11ea-836c-65c1ada2d766.gif)

https://github.com/zombieFox/nightTab/pull/186

**Comment by rezpower** [comment]
First of all a big thanks for this extension! It's great , intelligent, fully customizable and just perfect.
To get back to the topic: It would be also nice if you we could give a path on PC to bookmark images.
This way there is no need to get them from an online source or save them in addon. Just a folder on the PC where you store your personalized icons and favicons etc.

**Comment by zombieFox** [comment]
>  It would be also nice if you we could give a path on PC to bookmark images.

I'm pretty sure browser extensions are not able to access local system files for security reasons.

**Comment by rezpower** [comment]
what about only picking the file once from a local folder and saving it, just like when you pick a background picture?

**Comment by zombieFox** [comment]
Yes a file can be accessed via the [files type input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file) element, this is indeed how the background image upload works. 

However this solution has some major issues when working on a project like this, eg, no database and storing all data in browser [local storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage). 

Storage space is one of the issues, image support is another. Please see issue #69 for more details. Users have also encountered major bugs caused by large image date stored in the local storage causing nightTab updates to fail. It is a difficutl isses, one which I have not found a solution to yet (other than perhaps removing the background image upload feature all together [still considering this]).

Short version, it is safer and more predictable for the end user to use [external resources](https://github.com/zombieFox/nightTab/issues/170#issuecomment-622535791) accessed by URLs for assets not controled by nightTab.

**Comment by Excigma** [comment]
You could upload the icons to Imgur, or something similar and use that I guess

**Comment by rezpower** [comment]
Thanks a lot for all the info. Finally the damn extension is soooo good and customizable that you almost forget any need for thumbnails!
![Annotation 2020-06-10 024657](https://user-images.githubusercontent.com/28585954/84215438-86cc3d80-aac6-11ea-8d48-c9f78c18f0f9.png)

**Comment by zombieFox** [comment]
This feature is now on live 🤖
